src/Plugin/GraphQL/DataProducer/Taxonomy/TaxonomyLoadTree.php

```diff
@@ -110,7 +149,43 @@ public function resolve(string $vid, int $parent = 0, ?int $max_depth = NULL, Re
       ->getStorage('taxonomy_term')
       ->loadTree($vid, $parent, $max_depth);
 
-    return $terms;
+    $term_ids = array_column($terms, 'tid');
+    $resolver = $this->entityBuffer->add('taxonomy_term', $term_ids);
+
+    return new Deferred(function () use ($type, $ids, $language, $resolver, $context, $access, $accessUser, $accessOperation) {
```

Comment:
$language is undefined? Same for $ids? Please check all variables if they actually exist here.

Reply:
Fixed language usage, removed ids and removed todo. As we agreed, for our use case I introduce secondary data producer which just loads stdClass objects with language and access checks